pkg/core/interops.go

```diff
@@ -66,213 +66,215 @@ type interopedFunction struct {
 
 // getSystemInterop returns matching interop function from the System namespace
 // for a given id in the current context.
-func (ic *interopContext) getSystemInterop(id uint32) *vm.InteropFuncPrice {
-	return ic.getInteropFromSlice(id, systemInterops)
+func getSystemInterop(ic *interopContext) vm.InteropGetterFunc {
+	return getInteropFromSlice(ic, systemInterops)
 }
 
 // getNeoInterop returns matching interop function from the Neo and AntShares
 // namespaces for a given id in the current context.
-func (ic *interopContext) getNeoInterop(id uint32) *vm.InteropFuncPrice {
-	return ic.getInteropFromSlice(id, neoInterops)
+func getNeoInterop(ic *interopContext) vm.InteropGetterFunc {
+	return getInteropFromSlice(ic, neoInterops)
 }
 
 // getInteropFromSlice returns matching interop function from the given slice of
 // interop functions in the current context.
-func (ic *interopContext) getInteropFromSlice(id uint32, slice []interopedFunction) *vm.InteropFuncPrice {
-	n := sort.Search(len(slice), func(i int) bool {
-		return slice[i].ID >= id
-	})
-	if n < len(slice) && slice[n].ID == id {
-		// Currying, yay!
-		return &vm.InteropFuncPrice{Func: func(v *vm.VM) error {
-			return slice[n].Func(ic, v)
-		}, Price: slice[n].Price}
+func getInteropFromSlice(ic *interopContext, slice []interopedFunction) func(uint32) *vm.InteropFuncPrice {
+	return func(id uint32) *vm.InteropFuncPrice {
+		n := sort.Search(len(slice), func(i int) bool {
+			return slice[i].ID >= id
+		})
+		if n < len(slice) && slice[n].ID == id {
+			// Currying, yay!
```

Comment:
Well, no more currying here after `bad39dd8863e32fbfa64d0e9fac9f1ec3448c678` (you see, that why it starts with `bad`!), so the comment is no longer relevant.